tests/lib/rules/require-valid-default-prop.js

```
@@ -0,0 +1,378 @@
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: false
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: {}
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: []
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: ''
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: 55
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: false
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: {}
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: []
```

Did you think about supporting also:
```
{
  type: Array,
  default() {
    return 100
  }
}
```
? It should also throw an error I think